### PR TITLE
Bump to PHP 7.2, stick to phpunit 8.5

### PR DESCRIPTION
**Pull request by nicolas-grekas**
I think it's start to shade PHP 7.1 out. Of course the polyfill will be kept functional, but we won't care about it anymore from a maintenance PoV.

Sticking to phpunit 8.5 so that we can focus on one version of it, and because it has [lifetime support](https://phpunit.de/supported-versions.html) and PHP 7.2 as a minimum version.

### CODE REVIEW

**Comment by derrabus** [comment]
What has changed since #428?

From Symfony's PoV, taking into account that the active maintenance of the 5.4 branch will end about two months from now, we might as well wait a little longer and bump to PHP 8.1 directly.

File: src/Php72/composer.json [hunk 1]
```diff
@@ -1,32 +0,0 @@
...
-            "name": "Nicolas Grekas",
-            "email": "p@tchwork.com"
-        },
-        {
-            "name": "Symfony Community",
-            "homepage": "https://symfony.com/contributors"
-        }
-    ],
-    "require": {
-        "php": ">=7.1"
```
Comment (derrabus) [inline src/Php72/composer.json:19]: In #428, I did not remove the package just yet, but instead bumped it to PHP 7.2 as well and made it a no-op package. We've done this with other polyfills in the past as well, e.g. https://packagist.org/packages/symfony/polyfill-php71#v1.20.0

**Comment by nicolas-grekas** [comment]
I made this PR after realizing that #486 was required because on PHP 7.1 we have to use var-dumper 4.4, which didn't get the fixes for nullable arguments.
Then I also realized that phpunit 8.5 has lifetime support and also php >= 7.2.
The CI is behind at the moment with e.g. #425 + failures and I don't think it would be productive to work on this while also having to consider several versions of phpunit.

PHP 8.1 is too early IMHO, at least I don't see similar arguments.

File: src/Php72/composer.json (hunk 1 above)
Comment (nicolas-grekas) [reply src/Php72/composer.json:19]: why did we do this? what's the benefit?
people with php < 7.2 can still install a previous version

**Comment by derrabus** [comment]
If PHP 7.1 support really blocks our CI, go ahead then. But please, let's do one release of `polyfill-php72` as a no-op package as I proposed in https://github.com/symfony/polyfill/pull/492#discussion_r1749948439

> PHP 8.1 is too early IMHO, at least I don't see similar arguments.

Let's revisit that topic in December or January. 🙂

**Comment by nicolas-grekas** [comment]
> let's do one release of polyfill-php72 as a no-op package

why did we do this? what's the benefit?
people with php < 7.2 can still install a previous version

**Comment by derrabus** [comment]
> > let's do one release of polyfill-php72 as a no-op package
> 
> why did we do this? what's the benefit?

If you're on PHP >= 7.2, but one of your dependency's dependencies requires the polyfill, you get a package with zero overhead. Once we tagged a new release this way, we can still remove the package from the monorepo.

**Comment by nicolas-grekas** [comment]
Got it, thanks. PR updated.

**Review by derrabus — APPROVED** [review]

**Pull request merged**